src/main/java/io/cdap/plugin/batch/http/to/hdfs/HTTPToHDFSActionConfig.java

```diff
@@ -0,0 +1,407 @@
+    this.numRetries = numRetries;
+    this.readTimeout = readTimeout;
+    this.connectTimeout = connectTimeout;
+    this.followRedirects = followRedirects;
+    this.disableSSLValidation = disableSSLValidation;
+    this.outputPath = outputPath;
+    this.responseHeaders = responseHeaders;
+  }
+
+  private HTTPToHDFSActionConfig(Builder builder) {
+    hdfsFilePath = builder.hdfsFilePath;
+    url = builder.url;
+    requestHeaders = builder.requestHeaders;
+    outputFormat = builder.outputFormat;
+    charset = builder.charset;
+    method = builder.method;
+    body = builder.body;
+    numRetries = builder.numRetries;
+    readTimeout = builder.readTimeout;
+    connectTimeout = builder.connectTimeout;
+    followRedirects = builder.followRedirects;
+    disableSSLValidation = builder.disableSSLValidation;
+    outputPath = builder.outputPath;
+    responseHeaders = builder.responseHeaders;
+  }
+
+  public static Builder builder() {
+    return new Builder();
+  }
+
+  public static Builder builder(HTTPToHDFSActionConfig copy) {
+    return builder()
+      .setHdfsFilePath(copy.hdfsFilePath)
+      .setUrl(copy.url)
+      .setRequestHeaders(copy.requestHeaders)
+      .setOutputFormat(copy.outputFormat)
+      .setCharset(copy.charset)
+      .setMethod(copy.method)
+      .setBody(copy.body)
+      .setNumRetries(copy.numRetries)
+      .setReadTimeout(copy.readTimeout)
+      .setConnectTimeout(copy.connectTimeout)
+      .setFollowRedirects(copy.getFollowRedirects())
+      .setDisableSSLValidation(copy.disableSSLValidation)
+      .setOutputPath(copy.outputPath)
+      .setResponseHeaders(copy.responseHeaders);
+  }
+
+  public String getHdfsFilePath() {
+    return hdfsFilePath;
+  }
+
+  public String getUrl() {
+    return url;
+  }
+
+  public String getMethod() {
+    return method;
+  }
+
+  @Nullable
+  public String getBody() {
+    return body;
+  }
+
+  public String getOutputFormat() {
+    return outputFormat;
+  }
+
+  public String getCharset() {
+    return charset;
+  }
+
+  public Boolean getFollowRedirects() {
+    return followRedirects;
+  }
+
+  public Boolean getDisableSSLValidation() {
+    return disableSSLValidation;
+  }
+
+  public Integer getNumRetries() {
+    return numRetries;
+  }
+
+  @Nullable
+  public Integer getConnectTimeout() {
+    return connectTimeout;
+  }
+
+  @Nullable
+  public Integer getReadTimeout() {
+    return readTimeout;
+  }
+
+  @Nullable
+  public String getOutputPath() {
+    return outputPath;
+  }
+
+  @Nullable
+  public String getResponseHeaders() {
+    return responseHeaders;
+  }
+
+  public Map<String, String> getRequestHeadersMap() {
+    return convertHeadersToMap(requestHeaders);
+  }
+
+  public void validate(FailureCollector failureCollector) {
+    try {
+      new URL(url);
+    } catch (MalformedURLException e) {
+      failureCollector.addFailure(String.format("URL '%s' is malformed.", url), null)
+        .withStacktrace(e.getStackTrace())
```

Comment:
I don't think stack trace is necessary here. We possibly can add e.geMessage() to failure if it has some useful information, otherwise we don't need it at all.